linkerd/app/src/env.rs

```diff
@@ -800,29 +818,44 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
         .unwrap_or(super::tap::Config::Disabled);
 
     let identity = {
-        let (addr, certify, tls) = identity_config?;
-        // If the address doesn't have a server identity, then we're on localhost.
-        let connect = if addr.addr.is_loopback() {
-            inbound.proxy.connect.clone()
-        } else {
-            outbound.proxy.connect.clone()
-        };
-        let failfast_timeout = if addr.addr.is_loopback() {
-            inbound.http_request_queue.failfast_timeout
-        } else {
-            outbound.http_request_queue.failfast_timeout
-        };
-        identity::Config::Linkerd {
-            certify,
-            tls,
-            client: ControlConfig {
-                addr,
-                connect,
-                buffer: QueueConfig {
-                    capacity: DEFAULT_CONTROL_QUEUE_CAPACITY,
-                    failfast_timeout,
+        let tls = tls?;
+
+        match strings.get(ENV_IDENTITY_SPIRE_SOCKET)? {
+            Some(socket) => identity::Config::Spire {
+                tls,
+                client: spire::Config {
+                    socket_addr: std::sync::Arc::new(socket),
+                    backoff: parse_backoff(strings, IDENTITY_SPIRE_BASE, DEFAULT_SPIRE_BACKOFF)?,
                 },
             },
```

Comment:
I think in this case we we want to validate that the local_id is a valid SPIFFE id.

Reply:
I have added the check for SPIFFE scheme for now. Is that enough for the time being? 

I also wonder, should we make the ID::Uri be spiffe specific and follow the requirements outlined in the SPIFFE RFC  (like we do with the dns name) or we should keep it as a URI type?

Reply:
On one hand, it would be useful to have a static assertion that the value has been validated; on the other hand, we want to use these types for values read from peers, and I don't think anything about Linkerd's _TLS_ system actually cares about SPIFFE.

So for now, I think it's sufficient to handle this where we habe both the context of (1) the configured local ID and (2) how we're going to provision credentials, i.e. "when SPIRE is enabled, validate that the configured ID is a SPIFFE ID."